src/store.rs

```diff
@@ -140,26 +137,84 @@ impl fmt::Display for Error {
 /// Prefer these methods when constructing literal data or assembling program fragments in
 /// tests or during evaluation, etc.
 impl<F: LurkField> Store<F> {
-    pub fn nil(&mut self) -> Ptr<F> {
-        self.lurk_sym("nil")
+    pub fn expect_constants(&self) -> &NamedConstants<F> {
+        self.constants
+            .get()
+            .expect("Constants must have been set during instatiation")
+    }
```

Comment:
I can just drop the lazy behavior because it's not really lazy. Basically we pretty much always want those pointers initialized. This will simplify the code.

Reply:
I'm not sure that's a great idea. That forces a tax proportional to the number of symbols on every usage of the store. Even if that's not highly noticeable because of our current (limited) usage, it is problematic in principle and will either cost us later or restrain us from decisions that should be independent.

These constants are only used in evaluation and proving, so we shouldn't pay for creating them when a `Store` is only used to construct arbitrary data. We'll also likely need a similar mechanism to deal with symbols required by coprocessors. If we make this a monolithic requirement paid on creation of every store, the cost to use cases that don't require paying the whole (or any of) that cost will continually grow.

Reply:
[That's already the case on `master`](https://github.com/lurk-lab/lurk-rs/blob/cfa69b4e33f29a51e72437caf574dd6704640087/src/store.rs#L121-L123)